src/inner_product_round.rs

```diff
@@ -63,81 +65,101 @@ impl<'a> InnerProductRound<'a> {
     pub fn init(
         gi_base: Vec<RistrettoPoint>,
         hi_base: Vec<RistrettoPoint>,
-        g_base: RistrettoPoint,
+        g_base: Vec<RistrettoPoint>,
         h_base: RistrettoPoint,
         ai: Vec<Scalar>,
         bi: Vec<Scalar>,
-        alpha: Scalar,
+        alpha: Vec<Scalar>,
         y_powers: Vec<Scalar>,
         transcript: &'a mut Transcript,
         seed_nonce: Option<Scalar>,
         aggregation_factor: usize,
     ) -> Result<Self, ProofError> {
         let n = gi_base.len();
         if gi_base.is_empty() || hi_base.is_empty() || ai.is_empty() || bi.is_empty() || y_powers.is_empty() {
-            Err(ProofError::InvalidLength(
+            return Err(ProofError::InvalidLength(
                 "Vectors gi_base, hi_base, ai, bi and y_powers cannot be empty".to_string(),
-            ))
-        } else if !(hi_base.len() == n && ai.len() == n && bi.len() == n) || (y_powers.len() != (n + 2)) {
-            Err(ProofError::InvalidArgument(
+            ));
+        }
+        if !(hi_base.len() == n && ai.len() == n && bi.len() == n) || (y_powers.len() != (n + 2)) {
```

Comment:
Doesn't `gi_base` need its length checked as well?

Reply:
Indeed it does, equal to `n` for that statement (`let n = gi_base.len();`).